code/hud/hudtarget.cpp

```diff
@@ -1591,9 +1591,14 @@ int hud_target_ship_can_be_scanned(ship *shipp)
 	if (shipp->flags[Ship::Ship_Flags::Scannable])
 		return 1;
 
-	// ignore ships that don't carry cargo
-	if ((sip->class_type < 0) || !(Ship_types[sip->class_type].flags[Ship::Type_Info_Flags::Scannable]))
+	if (!Use_new_scanning_behavior) {
```

Comment:
Inverting the condition here would let this be unified with the previous check, making it easier to read.

```cpp
// The old behavior treats some ship types as always scannable. The new behavior only checks the ship's Scannable flag.
if (shipp->flags[Ship::Ship_Flags::Scannable])
    return 1;
} else if (Use_new_scanning_behavior) {
    return 0;
} else if ((sip->class_type < 0) || !(Ship_types[sip->class_type].flags[Ship::Type_Info_Flags::Scannable])) {
    return 0;
}
```

Reply:
Good call. Done.